config/deploy/qa.rb

```diff
@@ -4,3 +4,11 @@
 server 'argo-qa-b.stanford.edu', user: 'lyberadmin', roles: %w[web db app worker]
 
 Capistrano::OneTimeKey.generate_one_time_key!
+
+before 'deploy:starting', :ruby_version_check do
+  app_ruby = Bundler::LockfileParser.new(Bundler.read_file('Gemfile.lock')).ruby_version.split.last.split('p').first
+  # default_system_ruby = `rvm list default string`.chomp.chomp.split('-').last
+  system_rubies = `rvm list rubies`.split.grep(/ruby/).map { |version| version.delete_prefix('ruby-') }
```

Comment:
I'm not sure that looking for an installed ruby version is good enough. We need to know what passenger is using.  If the version in the Gemfile doesn't match what passenger has, I think it will break.

Reply:
that's in the list of things @aaron-collier is looking at. I wouldn't look too closely at this PR until it's out of draft.